src/test/rules/noInnerDeclarationsRuleTests.ts

```diff
@@ -16,7 +16,9 @@ const scripts = {
     'function decl(arg) { var fn; if (arg) { fn = function expr() { }; } }',
     'if (test) { var foo; }',
     'function doSomething() { while (test) { var foo; } }',
-    'foo(() => { function bar() { } });'
+    'foo(() => { function bar() { } });',
+    'namespace something { function decl(arg) { var foo; } }',
+    'class MyClass { constructor(arg) { function decl(x) { var foo; } } }'
```

Comment:
Should a class constructor allow this? It would make more sense to simply use an arrow function:

```typescript
class MyClass { constructor(arg) {const decl = (x) => { var foo; } } }'
```

Otherwise we can simply define `decl` as method and call it from the constructor. Is there any option to specify where the rule should apply. I wouldn't want to see this type of code in methods and constructors, perhaps this should be optional.